internal/services/datafactory/data_factory_linked_service_sftp_resource_test.go

```diff
@@ -144,46 +131,90 @@ resource "azurerm_data_factory_linked_service_sftp" "test" {
 }
-`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger)
+`, t.template(data), data.RandomInteger)
 }
 
-func (LinkedServiceSFTPResource) update2(data acceptance.TestData) string {
+func (t LinkedServiceSFTPResource) update2(data acceptance.TestData) string {
 	return fmt.Sprintf(`
-provider "azurerm" {
-  features {}
-}
+%s
+resource "azurerm_data_factory_linked_service_sftp" "test" {
+  name                     = "acctestlssftp%d"
+  data_factory_id          = azurerm_data_factory.test.id
+  authentication_type      = "Basic"
+  host                     = "http://www.bing.com"
+  port                     = 22
+  username                 = "foo"
+  password                 = "bar"
+  annotations              = ["test1", "test2"]
+  description              = "test description 2"
+  skip_host_key_validation = true
+  host_key_fingerprint     = "fingerprint"
 
-resource "azurerm_resource_group" "test" {
-  name     = "acctestRG-df-%d"
-  location = "%s"
-}
+  parameters = {
+    foo  = "test1"
+    bar  = "test2"
+    buzz = "test3"
+  }
 
-resource "azurerm_data_factory" "test" {
-  name                = "acctestdf%d"
-  location            = azurerm_resource_group.test.location
-  resource_group_name = azurerm_resource_group.test.name
+  additional_properties = {
+    foo = "test1"
+  }
+}
+`, t.template(data), data.RandomInteger)
 }
 
+func (t LinkedServiceSFTPResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
 resource "azurerm_data_factory_linked_service_sftp" "test" {
-  name                = "acctestlsweb%d"
-  data_factory_id     = azurerm_data_factory.test.id
-  authentication_type = "Basic"
-  host                = "http://www.bing.com"
-  port                = 22
-  username            = "foo"
-  password            = "bar"
-  annotations         = ["test1", "test2"]
-  description         = "test description 2"
+  name                     = "acctestlssftp%[2]d"
+  data_factory_id          = azurerm_data_factory.test.id
+  authentication_type      = "Basic"
+  host                     = "http://www.bing.com"
+  port                     = 22
+  username                 = "foo"
+  password                 = "bar"
+  annotations              = ["test1", "test2"]
+  description              = "test description 2"
+  skip_host_key_validation = true
+  host_key_fingerprint     = "fingerprint"
+  integration_runtime_name = azurerm_data_factory_integration_runtime_azure.test.name
 
   parameters = {
-    foo  = "test1"
-    bar  = "test2"
-    buzz = "test3"
+    foo = "test1"
+    bar = "test2"
   }
 
   additional_properties = {
     foo = "test1"
+    bar = "test1"
   }
 }
-`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger)
+
+resource "azurerm_data_factory_integration_runtime_azure" "test" {
+  data_factory_id = azurerm_data_factory.test.id
+  location        = azurerm_resource_group.test.location
+  name            = "acctestlssftp%[2]d"
+}
```

Comment:
The pattern that we use in test configs is to put the resource that's under test at the bottom of the config, so could you move this up and it would also be helpful to update the contributor docs with this information.

Reply:
Certainly, will do.

Reply:
How's this? https://github.com/hashicorp/terraform-provider-azurerm/pull/29010/files#diff-ad5cd26feeaf83afe3aeb28a503667fe6f2367239a5dd9f012e87ccb88fa25aaR15